plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthManager.java

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.client;
+
+import com.google.inject.Inject;
+
+import org.apache.commons.codec.binary.Base64;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for managing
+ *
+ * @author Mykola Morhun
+ */
+public class DockerRegistryAuthManager {
+
+    private static final String DEFAULT_REGISTRY = "";
+
+    private final InitialAuthConfig initialAuthConfig;
+
+    @Inject
+    public DockerRegistryAuthManager(InitialAuthConfig initialAuthConfig) {
+        this.initialAuthConfig = initialAuthConfig;
+    }
+
+    public String getDefaultUsername() {
+        return initialAuthConfig.getDefaultUsername();
+    }
+
+    /**
+     * Looks for auth header for specified registry.
+     * First searches in the params and then in the initial auth config.
+     * If nothing found empty json will be returned.
```

Comment:
Here javadocs says empty json will be returned. Nothing about encoding. It is possible that devs will not read return description. So it is better to duplicate explanation of base 64 encoding in the main part of javadocs